kotlin/spatial-operations/src/main/java/com/esri/arcgisruntime/sample/spatialoperations/MainActivity.kt

```diff
@@ -0,0 +1,217 @@
+  private var differenceMenuItem: MenuItem? = null
+  private var symmetricDifferenceMenuItem: MenuItem? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // create ArcGISMap with topographic basemap
+      map = ArcGISMap(Basemap.createLightGrayCanvas())
+
+      // create graphics overlays to show the inputs and results of the spatial operation
+      graphicsOverlays.add(inputGeometryOverlay)
+      graphicsOverlays.add(resultGeometryOverlay)
+
+      // create input polygons and add graphics to display these polygons in an overlay
+      createPolygons()
+
+      // center the map view on the input geometries
+      val envelope = GeometryEngine.union(inputPolygon1, inputPolygon2).extent
+      setViewpointGeometryAsync(envelope, 20.0)
+    }
+
+  }
+
+  private fun showGeometry(resultGeometry: Geometry) {
+    // add a graphic from the result geometry, showing result in red (0xFFE91F1F)
+    Graphic(resultGeometry, resultFillSymbol).let {
+      resultGeometryOverlay.graphics.add(it)
+
+      // select the result to highlight it
+      it.isSelected = true
+    }
+  }
+
+  private fun createPolygons() {
+    // create input polygon 1
+    PointCollection(SpatialReferences.getWebMercator()).run {
+      add(Point(-13160.0, 6710100.0))
+      add(Point(-13300.0, 6710500.0))
+      add(Point(-13760.0, 6710730.0))
+      add(Point(-14660.0, 6710000.0))
+      add(Point(-13960.0, 6709400.0))
+      inputPolygon1 = Polygon(this)
+    }
+
+    // create and add a blue graphic to show input polygon 1
+    SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, Color.BLUE, lineSymbol).let {
+      inputGeometryOverlay.graphics.add(Graphic(inputPolygon1, it))
+    }
+
+    // outer ring
+    val outerRing = Part(PointCollection(SpatialReferences.getWebMercator()).apply {
+      add(Point(-13060.0, 6711030.0))
+      add(Point(-12160.0, 6710730.0))
+      add(Point(-13160.0, 6709700.0))
+      add(Point(-14560.0, 6710730.0))
+      add(Point(-13060.0, 6711030.0))
+    })
+
+    // inner ring
+    val innerRing = Part(PointCollection(SpatialReferences.getWebMercator()).apply {
+      add(Point(-13060.0, 6710910.0))
+      add(Point(-12450.0, 6710660.0))
+      add(Point(-13160.0, 6709900.0))
+      add(Point(-14160.0, 6710630.0))
+      add(Point(-13060.0, 6710910.0))
+    })
+
+    // add both parts (rings) to a part collection and create a geometry from it
+    PartCollection(outerRing).run {
+      add(innerRing)
+      inputPolygon2 = Polygon(this)
+    }
+
+    // create and add a green graphic to show input polygon 2
+    SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, -0x66ff6700, lineSymbol).let {
+      inputGeometryOverlay.graphics.add(Graphic(inputPolygon2, it))
+    }
+
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+    // Inflate the menu; this adds items to the action bar if it is present.
```

Comment:
this is super nit picky, but just for consistency in style across samples, all comments should follow one style. e.g. the majority of comments in this sample start off lower case, whereas there are some in this method which start off sentence case.